2230/3/rap-battle/rap-battle.rb

```diff
@@ -0,0 +1,160 @@
+require 'trollop' # Trollop is a commandline option parser for Ruby
+require 'russian_obscenity' # Gem for filtering russian obscene language
+require './utils' # Utility functions
+
+opts = Trollop.options do
+  opt %s[top-bad-words], 'Prints <i> records from the Top Bad Words', default: 0
+  # opt %s[top-words], 'Prints <i> words per people or list rappers if run without --name', :default => 30
+  # opt :name, 'Name of the rapper. Use with --top-words', :type => :string
+end
+
+# rules for ARGV parametrs
+# Trollop::educate if ARGV.empty?
+Trollop.die %s[top-bad-words], 'must be non-negative' if opts[%s[top-bad-words]] <= 0
+# p opts
+
+class FileReader
+  attr_reader :content
+
+  def initialize
+    @content = ''
+  end
+
+  def read(files)
+    @content = get_all_text(files)
+  end
+
+  # :reek:FeatureEnvy
+  # :reek:NestedIterators
+  # :reek:TooManyStatements
+  def get_all_text(files)
+    content = ''
+    files.each do |file_name|
+      fh = open "data/#{file_name.gsub(/:[()"'*.-]/) { |sym| '\\' + sym }.chomp}"
+      content += fh.read
+
+      fh.close
+    end
+    content
+  end
+end
+
+class DataLoader
+  attr_reader :data
+
+  def initialize
+    @data = [] # public
+    @rapper_names = [] # private
+  end
+
+  def load_data
+    @rapper_names = load_rapper_names
+    @data = build_data
+  end
+
+  private
+
+  def load_rapper_names
+    fh = open 'config/rapper_names'
+    fh.map(&:rstrip)
+  end
+
+  # Generate list of files
+  # rubocop:disable Style/InverseMethods
+  def build_file_list(folder)
+    Dir.entries(folder).select { |file_name| !File.directory? file_name }
+  end
+  # rubocop:enable Style/InverseMethods
+
+  # parse list of files
+  # :reek:TooManyStatements
+  def get_file_list(name)
+    common_file_list = build_file_list('data')
+    file_list = []
+    # filter rapper names by name
+    names = Utils.escape_string(@rapper_names.grep(/#{Utils.escape_string(name)}/)[0].gsub(', ', '|'))
+    common_file_list.grep(/#{names}/i).each do |file_name|
+      file_list << "#{file_name}\n" if file_name.sub(/\s(прот|vs|VS ).+/, '').index(/#{names}/)
+    end
+    file_list
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/AbcSize
+  # :reek:TooManyStatements
+  def build_data
+    @rapper_names.each do |rapper_name|
+      rapper_name = rapper_name.sub(/,.*/, '')
```

Comment:
prepare good data for rapper_names

BTW, you should use sub! instead of sub for this in future.